File: frontend/src/utils/outfitPromptUtils.ts
import type { OutfitSuggestion } from '../models/OutfitModels';

const MAX_PREVIOUS_OUTFIT_CHARS = 6000;

/**
 * Serialize the current outfit for the API so the model can suggest a clearly different alternative.
 */
export function formatPreviousOutfitForPrompt(s: OutfitSuggestion): string {
  const lines = [
    `Shirt: ${s.shirt}`,
    `Trousers: ${s.trouser}`,
    `Blazer: ${s.blazer}`,
    `Shoes: ${s.shoes}`,
    `Belt: ${s.belt}`,
    `Reasoning: ${s.reasoning}`,
  ];
  const text = lines.join('\n');
  if (text.length <= MAX_PREVIOUS_OUTFIT_CHARS) {
    return text;
  }
  return `${text.slice(0, MAX_PREVIOUS_OUTFIT_CHARS)}\n[truncated]`;
}

File: ios-client/OutfitSuggestor/Utils/OutfitPromptUtils.swift
//
//  OutfitPromptUtils.swift
//  OutfitSuggestor
//
//  Serialize outfits for API variety prompts — keep in sync with outfitPromptUtils.ts
//

import Foundation

enum OutfitPromptUtils {
    private static let maxPreviousOutfitChars = 6000

    static func formatPreviousOutfitForPrompt(_ suggestion: OutfitSuggestion) -> String {
        let lines = [
            "Shirt: \(suggestion.shirt)",
            "Trousers: \(suggestion.trouser)",
            "Blazer: \(suggestion.blazer)",
            "Shoes: \(suggestion.shoes)",
            "Belt: \(suggestion.belt)",
            "Reasoning: \(suggestion.reasoning)",
        ]
        let text = lines.joined(separator: "\n")
        if text.count <= maxPreviousOutfitChars {
            return text
        }
        let prefix = String(text.prefix(maxPreviousOutfitChars))
        return "\(prefix)\n[truncated]"
    }
}
